solver_spec.rb

```diff
@@ -0,0 +1,54 @@
+require_relative 'solver'
+require 'rspec'
+
+RSpec.describe Solver do
+  let(:solver) { Solver.new }
+
+  describe '#factorial' do
+    context 'Test the factorial for non-negative numbers' do
+      it 'returns the factorial of the given number' do
+        expect(solver.factorial(5)).to eq(120)
+        expect(solver.factorial(0)).to eq(1)
+      end
+    end
+
+    context 'Test the factorial of a negative input' do
+      it 'exception error' do
+        expect { solver.factorial(-1) }.to raise_error(ArgumentError, 'Input must be a positive number!')
+      end
+    end
```

Comment:
- [ ] This particular test fails because the `Exception` you're raising is not exactly as you specified in the `Solver` class.
**Tip:** 
```ruby
expect { solver.factorial(-1) }.to raise_error(ArgumentError, 'This method only accepts 0 and positive integers')
```

OR

```ruby
expect { solver.factorial(-1) }.to raise_error(ArgumentError)
```
will solve the problem 😎

Reply:
Thanks for the tip